### ISSUE TITLE
arg/3: Syntax error triggers crash

### ISSUE THREAD

**Issue by flexoron** [infradig/trealla#426]
```
?- arg(2,a(a,b,c),b).
   true.
?- arg(2,a(a,b,c),).
Segmentation fault (core dumped)
```

**Comment by infradig** [comment]
?- X=f(2,a(a,b,c),).
   X = f(2,a(a,b,c),false).

it's not picking up on a parsing error.

?- f(2,a(a,b,c),).
uncaught exception: error(existence_error(procedure,f/3),f/3)

That should have got a syntax error.

**Comment by flexoron** [comment]
v1.20.22
```
?- [user].
f(_,_,X) :- X = false.
   true.
?- f(_,_,).
Segmentation fault (core dumped)
```
fixed with v1.20.23
```
?- [user].
f(_,_,X) :- X = false.
   true.
?- f(_,_,).
Error: syntax error, operator expected, line 1: ,), ''
?- f(_,_,false).
   true.
```
